test/SocketTest.php

```diff
@@ -23,4 +23,18 @@ public function testReadAndClose() {
             }
         });
     }
+
+    public function testLocalAddressAsUnixSocket() {
+        @unlink(__DIR__ . '/socket.sock');
+
+        $socket = \socket_create(AF_UNIX, SOCK_STREAM, 0);
+        \socket_bind($socket, __DIR__ . '/socket.sock');
+
+        socket_set_nonblock($socket);
+        socket_listen($socket);
+
+        $clientSocket = new Socket\ClientSocket(stream_socket_client('unix://' . __DIR__ . '/socket.sock'));
```

Comment:
Please prepend a NUL-byte to the path, then it also doesn't need the `.gitignore` entry.